include/tvm/runtime/container.h

```diff
@@ -274,6 +275,99 @@ class ADT : public ObjectRef {
   TVM_DEFINE_OBJECT_REF_METHODS(ADT, ObjectRef, ADTObj);
 };
 
+/*! \brief An object representing string. It's POD type. */
+class StringObj : public Object {
+ public:
+  /*! \brief The length of the string object. */
+  uint32_t size;
+
+  /*! \brief The pointer to string data. */
+  char* data;
+
+  /*! \brief String object which is moved from std::string container. */
+  class FromStd;
+};
+
+/*! \brief An object representing string moved from std::string. */
+class StringObj::FromStd : public StringObj {
```

Comment:
Let us move fromStd to "implementation section in the end", as they are not supposed to be seen by the user